querydsl-mongodb/src/main/java/com/querydsl/mongodb/MongodbSerializer.java

```diff
@@ -118,19 +119,30 @@ public Object visit(Operation<?> expr, Void context) {
         } else if (op == Ops.NOT) {
             //Handle the not's child
             BasicDBObject arg = (BasicDBObject) handle(expr.getArg(0));
-
-            //Only support the first key, let's see if there
-            //is cases where this will get broken
-            String key = arg.keySet().iterator().next();
             Operation<?> subOperation = (Operation<?>) expr.getArg(0);
             Operator subOp = subOperation.getOperator();
+
             if (subOp == Ops.IN) {
                 return visit(OperationImpl.create(Boolean.class, Ops.NOT_IN, subOperation.getArg(0),
                         subOperation.getArg(1)), context);
-            } else if (subOp != Ops.EQ && subOp != Ops.STRING_IS_EMPTY) {
-                return asDBObject(key, asDBObject("$not", arg.get(key)));
+            } else if (arg.size() == 1) {
+                Map.Entry<String, Object> entry = arg.entrySet().iterator().next();
+                if (entry.getKey().equals("$or")) {
+                    return asDBObject("$nor", entry.getValue());
+                } else if (entry.getKey().equals("$and")) {
+                    return asDBObject("$nand", entry.getValue());
```

Comment:
Relevant SO post:
http://stackoverflow.com/questions/20810698/mongodb-how-to-query-with-nand-operator.